src/parse.c

```diff
@@ -10255,6 +10264,139 @@ static psl_node_t p_psl_assert_directive(void)
    psl_node_t a = psl_new(P_ASSERT);
    psl_set_value(a, p);
 
+   if (peek() == tREPORT)
+      p_psl_report(a);
+
+   consume(tSEMI);
+
+   psl_set_loc(a, CURRENT_LOC);
+   return a;
+}
+
+static psl_node_t p_psl_assume_directive(void)
+{
+   // assume Property ;
+   // assume_guarantee Property [ report String ] ;
+
+   BEGIN("assume directive");
+
+   token_t tok = peek();
+   assert(tok == tASSUME || tok == tASSUMEG);
+   consume(tok);
+
+   psl_node_t a = psl_new(P_ASSUME);
+   if (tok == tASSUME)
+      psl_set_subkind(a, PSL_NO_GUARANTEE);
+   else
+      psl_set_subkind(a, PSL_GUARANTEE);
+
+   psl_node_t p = p_psl_property();
+   if (!psl_has_clock(p))
+      psl_set_clock(p, find_default_clock(nametab));
+   psl_set_value(a, p);
+
+   if (peek() == tREPORT && tok == tASSUMEG)
+      p_psl_report(a);
+
+   consume(tSEMI);
+
+   psl_set_loc(a, CURRENT_LOC);
+   return a;
+}
+
+static psl_node_t p_psl_restrict_directive(void)
+{
+   // restrict Sequence  ;
+   // restrict_guarantee Sequence [ report String ] ;
+
+   BEGIN("restrict directive");
+
+   token_t tok = peek();
+   assert(tok == tRESTRICT || tok == tRESTRICTG);
+   consume(tok);
+
+   psl_node_t a = psl_new(P_RESTRICT);
+   if (tok == tRESTRICT)
+      psl_set_subkind(a, PSL_NO_GUARANTEE);
+   else
+      psl_set_subkind(a, PSL_GUARANTEE);
+
+   psl_node_t p = p_psl_sequence();
+   if (!psl_has_clock(p))
+      psl_set_clock(p, find_default_clock(nametab));
+   psl_set_value(a, p);
+
+   if (peek() == tREPORT && tok == tRESTRICTG)
+      p_psl_report(a);
+
+   consume(tSEMI);
+
+   psl_set_loc(a, CURRENT_LOC);
+   return a;
+}
+
+static psl_node_t p_psl_fairness(void)
+{
+   // fairness Boolean ;
+   // strong fairness Boolean , Boolean ;
+
+   BEGIN("fairness statement");
+
+   psl_strength_t strength = PSL_WEAK;
+   if (peek() == tSTRONG) {
+      consume(tSTRONG);
+      strength = PSL_STRONG;
+   }
+
+   consume(tFAIRNESS);
+
+   psl_node_t a = psl_new(P_FAIRNESS);
+   psl_set_subkind(a, strength);
+
+   tree_t e1 = p_expression();
+   solve_types(nametab, e1, std_type(NULL, STD_BOOLEAN));
```

Comment:
```suggestion
   solve_psl_condition(nametab, e1);
```

I think PSL Booleans can be `bit` and `std_logic` too. `solve_psl_condition` allows the implicit `??` condition conversion for VHDL-2008 or else `bit` and `std_logic` directly for earlier standards.